### PR TITLE
sequence_task_3_AMarmusevich

### PR DESCRIPTION
**Pull request by marmusevich**
[==========] Running 4 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 4 tests from Reverse
[ RUN      ] Reverse.Easy
[       OK ] Reverse.Easy (1 ms)
[ RUN      ] Reverse.MultipleSpaces
[       OK ] Reverse.MultipleSpaces (0 ms)
[ RUN      ] Reverse.TrailingSpaces
[       OK ] Reverse.TrailingSpaces (0 ms)
[ RUN      ] Reverse.Corner
[       OK ] Reverse.Corner (0 ms)
[----------] 4 tests from Reverse (1 ms total)

[----------] Global test environment tear-down
[==========] 4 tests from 1 test suite ran. (2 ms total)
[  PASSED  ] 4 tests.

### CODE REVIEW

File: sequence_task_3/student/src/impl.cpp [hunk 1]
```diff
@@ -1,2 +1,21 @@
 #include "impl.h"
 
+std::string reverseWords(const std::string& input)
+{
+    std::string ret;
+    if(!input.empty())
+    {
+        std::stringstream in(input);
+        const std::deque<std::string> ds { std::istream_iterator<std::string>(in), std::istream_iterator<std::string>() };
```
Comment (smay1613) [inline sequence_task_3/student/src/impl.cpp:9]: Why deque?

Comment (marmusevich) [reply sequence_task_3/student/src/impl.cpp:9]: vector preferred, but task about deque

File: sequence_task_3/student/src/impl.cpp [hunk 2]
```diff
@@ -1,2 +1,23 @@
 #include "impl.h"
 
+#include <vector>
+
+std::string reverseWords(const std::string& input)
+{
+    std::string ret;
```
Comment (smay1613) [inline sequence_task_3/student/src/impl.cpp:7]: please use appropriate naming and don't use abbreviations

File: sequence_task_3/student/src/impl.cpp [hunk 3]
```diff
@@ -1,2 +1,23 @@
...
+    if(!input.empty())
```
Comment (smay1613) [inline sequence_task_3/student/src/impl.cpp:8]: Will be much simpler:
```suggestion
    if (input.empty()) {
        return input;
    }
```

**Pull request closed without merging**